### DISCUSSION TITLE
Issues with Calling public.update_user Function in Supabase via Postman and FlutterFlow

### DISCUSSION THREAD

**Discussion by Matignaciom** [supabase/supabase#28379] [Questions] [answered]
Hello community,

I am encountering some issues when trying to call the public.update_user function in my Supabase project. Here are the details:

**Problem in Postman**:

When making a call to the public.update_user function, I receive the following error:

```
{
    "code": "PGRST202",
    "details": "Searched for the function public.update_user with parameters direccion, fecha_nacimiento, id, nombre, telefono or with a single unnamed json/jsonb parameter, but no matches were found in the schema cache.",
    "hint": "Perhaps you meant to call the function public.update_user(p_direccion, p_fecha_nacimiento, p_id, p_nombre, p_telefono)",
    "message": "Could not find the function public.update_user(direccion, fecha_nacimiento, id, nombre, telefono) in the schema cache"
}
```

It seems that the function cannot find the correct definition in the schema cache. How can I resolve this?

**Problem in FlutterFlow API Call**:

When attempting an API call from FlutterFlow, I get the following error:

```
{
    "code": "42501",
    "details": null,
    "hint": null,
    "message": "permission denied for table users"
}
```

This indicates that I do not have permission to access the usuarios table. What permissions do I need to configure for this to work?

**Current Function Definition**:

Here is the current definition of the function in Supabase:

```
CREATE OR REPLACE FUNCTION public.update_user(
    p_id UUID,
    p_nombre TEXT DEFAULT NULL,
    p_direccion TEXT DEFAULT NULL,
    p_telefono TEXT DEFAULT NULL,
    p_fecha_nacimiento TEXT DEFAULT NULL
) RETURNS void AS $$
BEGIN
    -- Check if the user exists
    IF EXISTS (SELECT 1 FROM usuarios WHERE id = p_id) THEN
        -- Update the provided fields
        UPDATE usuarios
        SET 
            nombre = COALESCE(p_nombre, nombre),
            direccion = COALESCE(p_direccion, direccion),
            telefono = COALESCE(p_telefono, telefono),
            fecha_nacimiento = COALESCE(NULLIF(p_fecha_nacimiento, '')::date, fecha_nacimiento)
        WHERE id = p_id;
    ELSE
        RAISE EXCEPTION 'User with ID % not found', p_id;
    END IF;
END;
$$ LANGUAGE plpgsql;
```

Could someone help me resolve these issues or suggest how I should adjust the configuration or the function?

**Thank you in advance for any assistance!**

**Comment by GaryAustin1** [comment, marked as answer]
Your error seems to be you are calling the function with parameter names without the p_ in front of them in the rpc call.
![image](https://github.com/user-attachments/assets/b767d254-1a98-41d9-9aa4-d24c154b5da4)

You would need to present more info on your users table and how it was set up and if you are using something like Prisma which can remove permissions from the public schema.

**Reply by Matignaciom** [reply]
Hi! I was able to resolve this a while ago. The issue was that I needed to include the security definition in the function, and it worked perfectly. Thank you very much!
